nntrainer/layers/output_layer.cpp

```diff
@@ -52,6 +52,7 @@ void OutputLayer::calcDerivative() {
 
   Tensor &ret = net_input[0]->getGradientRef();
 
+  ret.setZero();
   for (unsigned int idx = 0; idx < getNumOutputs(); ++idx) {
     ret.add_i(net_hidden[idx]->getGradientRef());
```

Comment:
ret.add(..., ret) for idx == 0, and ret.add_i(...) for idx > 0?

Reply:
Updated. @lhs8928, please confer
```cpp
    if (idx == 0) {
      ret.fill(net_hidden[idx]->getGradientRef(), false);
    } else {
      ret.add_i(net_hidden[idx]->getGradientRef());
    }
```